src/Model/Filter/Boolean.php

```diff
@@ -25,7 +25,12 @@ public function process()
             return;
         }
 
-        $value = strtolower($this->value());
+        $value = $this->value();
+        if (!is_scalar($value)) {
+            return;
+        }
+
+        $value = strtolower($value);
```

Comment:
I think these options are good enough:
```

    protected $_defaultConfig = [
        'truthy' => [1, true, '1', 'true', 'yes', 'on'],
        'falsy' => [0, false, '0', 'false', 'no', 'off']
    ];
```
You can always extend those to also allow 'FALSE' or "OFF".

In case you really want to become case insensitive out of the box, this might work:
```php
if (is_string($value)) } {
  $value = strtolower($value);
}
```